src/algebra/algebra/tower.lean

```diff
@@ -307,6 +307,16 @@ le_antisymm (span_le.2 $ λ x hx, let ⟨p, q, hps, hqt, hpqx⟩ := set.mem_smul
   (λ _ _, add_mem _)
   (λ k x hx, smul_mem_span_smul' hs hx)
 
+lemma map_mem_span_algebra_map_image {S T : Type*} [comm_semiring S] [semiring T]
+  [algebra R S] [algebra R T] [algebra S T] [is_scalar_tower R S T]
+  (x : S) (a : set S) (hx : x ∈ submodule.span R a) :
+  algebra_map S T x ∈ submodule.span R (algebra_map S T '' a) :=
+begin
+  rw [← algebra.coe_linear_map, ← linear_map.coe_restrict_scalars R, ← submodule.map_span],
+  exact submodule.mem_map_of_mem hx,
+  all_goals { apply_instance }
```

Comment:
```suggestion
  rw [← algebra.coe_linear_map, ←(algebra.linear_map S T).coe_restrict_scalars R,
    ← submodule.map_span],
  exact submodule.mem_map_of_mem hx,
```

Reply:
Is it worth having something like these two lemmas instead?
```
/-- A variant of `submodule.span_image` for `algebra_map`. -/
lemma submodule.span_algebra_map_image {R A} [comm_semiring R] [semiring A] [algebra R A]
  (a : set R) :
  submodule.span R (algebra_map R A '' a) =
    (submodule.span R a).map (algebra.linear_map R A) :=
(submodule.span_image $ algebra.linear_map R A).trans rfl

lemma submodule.span_algebra_map_image_of_tower {S T : Type*} [comm_semiring S] [semiring T]
  [module R S] [is_scalar_tower R S S] [algebra R T] [algebra S T] [is_scalar_tower R S T]
  (x : S) (a : set S) :
  submodule.span R (algebra_map S T '' a) =
    (submodule.span R a).map ((algebra.linear_map S T).restrict_scalars R) :=
(submodule.span_image $ (algebra.linear_map S T).restrict_scalars R).trans rfl
```

Reply:
I'm guessing your silent resolution @Vierkantor means either "no, probably not" or "oops I didn't see that comment"

Reply:
The latter, thanks for pointing it out!